src/flash.ts

```diff
@@ -0,0 +1,58 @@
+import { format } from 'util'
+import { FastifyRequest, FastifyReply } from 'fastify'
+import { IncomingMessage, ServerResponse } from 'http'
+
+export interface ExtendedRequest<T = IncomingMessage> extends FastifyRequest<T> {
+  session: {
+    flash: {
+      [k: string]: string[]
+    }
+  }
+  flash: ReturnType<typeof flashFactory>['request']
+}
+
+export interface ExtendedReply<T = ServerResponse> extends FastifyReply<T> {
+  flash: ReturnType<typeof flashFactory>['reply']
+}
+
+export function flashFactory() {
+  let currentSession
+
+  return {
+    request(type: string, ...message: string[] | [string[]]): number {
+      if (message.length === 0) {
+        throw new Error('Provide a message to flash.')
+      }
+      if (Array.isArray(message[0])) {
+        message[0].forEach(msg => {
+          ;(currentSession.flash[type] = currentSession.flash[type] || []).push(msg)
+        })
+        return currentSession.flash[type].length
+      }
+
+      ;(currentSession.flash[type] = currentSession.flash[type] || []).push(
+        message.length > 1 ? format.apply(undefined, message) : message[0],
+      )
+
+      return currentSession.flash[type].length
+    },
+
+    reply(type?: string): typeof currentSession.flash | string[] {
+      if (!type) {
+        const allMessages = currentSession.flash
+        currentSession.flash = {}
+        return allMessages
+      }
+
+      const messages = currentSession.flash[type]
+      delete currentSession.flash[type]
```

Comment:
setting to `undefined`  is better.

Reply:
Since it is linked with session, if we set it to undefined the key will be still be present inside the session object. Is this really what we want?

Reply:
I think the session is serialized using `JSON.stringify()`, so those values will not be sent over the wire.